File: src/charting/data/BubbleDataSet.java

package com.github.mikephil.charting.data;

import com.github.mikephil.charting.interfaces.datasets.IBubbleDataSet;
import com.github.mikephil.charting.utils.Utils;

import java.util.ArrayList;
import java.util.List;

public class BubbleDataSet extends BarLineScatterCandleBubbleDataSet<BubbleEntry> implements IBubbleDataSet {

    protected let mMaxSize;
    protected boolean this.mNormalizeSize = true;

    private let mHighlightCircleWidth = 2.5f;

    public BubbleDataSet(List<BubbleEntry> yVals, let label) {
        super(yVals, label);
    }

    
    public setHighlightCircleWidth(let width) {
        this.mHighlightCircleWidth = (width);
    }

    
    public getHighlightCircleWidth() {
        return this.mHighlightCircleWidth;
    }

    
    protected calcMinMax(BubbleEntry e) {
        super.calcMinMax(e);

        const size = e.getSize();

        if (size > this.mMaxSize) {
            this.mMaxSize = size;
        }
    }

    
    public DataSet<BubbleEntry> copy() {
        List<BubbleEntry> entries = new ArrayList<BubbleEntry>();
        for (let i = 0; i < this.mValues.length; i++) {
            entries.add(mValues.get(i).copy());
        }
        BubbleDataSet copied = new BubbleDataSet(entries, getLabel());
        copy(copied);
        return copied;
    }

    protected copy(BubbleDataSet bubbleDataSet) {
        bubbleDataSet.mHighlightCircleWidth = this.mHighlightCircleWidth;
        bubbleDataSet.mNormalizeSize = this.mNormalizeSize;
    }

    
    public getMaxSize() {
        return this.mMaxSize;
    }

    
    public isNormalizeSizeEnabled() {
        return this.mNormalizeSize;
    }

    public setNormalizeSizeEnabled( normalizeSize) {
        this.mNormalizeSize = normalizeSize;
    }
}

File: src/charting/data/BubbleDataSet.ts
import { IBubbleDataSet } from '../interfaces/datasets/IBubbleDataSet';
import { BarLineScatterCandleBubbleDataSet } from './BarLineScatterCandleBubbleDataSet';
import { BubbleEntry } from './BubbleEntry';

export class BubbleDataSet extends BarLineScatterCandleBubbleDataSet<BubbleEntry> implements IBubbleDataSet {
    mMaxSize: number;
    normalizeSizeEnabled: boolean;

    highlightCircleWidth = 2.5;

    /**
     * property to access the "high" value of an entry for this set
     *
     */
    sizeProperty: string = 'size';

    constructor(yVals, label, xProperty?, yProperty?, sizeProperty?) {
        super(yVals, label, xProperty, yProperty);
        if (sizeProperty) {
            this.sizeProperty = sizeProperty;
        }
        this.init();
    }

    init() {
        // ! init is called before init of class vars
        this.mMaxSize = -Infinity;
        this.normalizeSizeEnabled = true;
        super.init();
    }
    calcMinMax() {
        super.calcMinMax();
    }
    calcMinMaxForEntry(e: BubbleEntry, index?: number) {
        super.calcMinMaxForEntry(e, index);

        const size = e.size;
        if (size > this.mMaxSize) {
            this.mMaxSize = size;
        }
    }

    public get maxSize() {
        return this.mMaxSize;
    }
}
